### PR TITLE
Fix various typos

### PR DESCRIPTION
**Pull request by Blake-Madden**
Misspellings and minor grammar issues

### CODE REVIEW

**Comment by American-Dipper** [comment]
@TylerMSFT - can you review the proposed changes?  These are mostly oblivously correct, but you might check and confirm the changes in these files:

- collections-c-cx.md
- platform-collections-unorderedmap-class.md
- weak-references-and-breaking-cycles-c-cx.md
 
IMPORTANT: When the changes are ready for publication, add a `#sign-off` comment to signal that the PR is ready for the review team to merge.  

 

#label:"aq-pr-triaged" 

@MicrosoftDocs/public-repo-pr-review-team

File: docs/code-quality/walkthrough-analyzing-c-cpp-code-for-defects.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ In this walkthrough, you'll:
 
      C6230: Implicit cast between semantically different types: using HRESULT in a Boolean context.
 
-     The code editor displays the line that caused the warning inside the function `bool ProcessDomain()`. This warning indicates that a `HRESULT` is being used in an 'if' statement where a Boolean result is expected. It's typically a mistake, because when the `S_OK` HRESULT is returned from a function it indicates success, but when converted into a boolean value it evaluates to **`false`**.
+     The code editor displays the line that caused the warning inside the function `bool ProcessDomain()`. This warning indicates that an `HRESULT` is being used in an 'if' statement where a Boolean result is expected. It's typically a mistake, because when the `S_OK` HRESULT is returned from a function it indicates success, but when converted into a boolean value it evaluates to **`false`**.
```
Comment (TylerMSFT) [inline docs/code-quality/walkthrough-analyzing-c-cpp-code-for-defects.md:93]: This one is wrong. It's 'a` `HRESULT`.  The rule is to use `a` if the h is pronounced like a consonant, which it is.  H, Result.

File: docs/cppcx/collections-c-cx.md [hunk 2]
```diff
@@ -12,9 +12,9 @@ The Windows Runtime defines the interfaces for collections and related types, an
 
 ![Diagram of C plus plus C X inheritance tree for collection types.](../cppcx/media/cppcxcollectionsinheritancetree.png)
 
-- The [Platform::Collections::Vector class](../cppcx/platform-collections-vector-class.md) resembles the [std::vector class](../standard-library/vector-class.md).
+- The [Platform::Collections::Vector Class](../cppcx/platform-collections-vector-class.md) resembles the [std::vector class](../standard-library/vector-class.md).
```
Comment (TylerMSFT) [inline docs/cppcx/collections-c-cx.md:15]: This one is wrong. Our style is to use sentence casing. Class should be lowercase.

File: docs/cppcx/collections-c-cx.md [hunk 3]
```diff
@@ -12,9 +12,9 @@ The Windows Runtime defines the interfaces for collections and related types, an
...
 
-- The [Platform::Collections::Map Class](../cppcx/platform-collections-map-class.md) class resembles the [std::map class](../standard-library/map-class.md).
+- The [Platform::Collections::Map Class](../cppcx/platform-collections-map-class.md) resembles the [std::map class](../standard-library/map-class.md).
```
Comment (TylerMSFT) [inline docs/cppcx/collections-c-cx.md:17]: To be consistent, we should lowercase Class in the brackets. Applies in other areas you've made changes if you want to make the same update. Not critical.

File: docs/ide/add-interface-definition-library-method-wizard.md [hunk 4]
```diff
@@ -42,7 +42,7 @@ The following section describes the wizard interface that you'll use to add a me
 
 - **Return type**
 
-  The data type returned by the method. The standard way to return error codes from methods defined in an interface is with a `HRESULT`.
+  The data type returned by the method. The standard way to return error codes from methods defined in an interface is with an `HRESULT`.
```
Comment (TylerMSFT) [inline docs/ide/add-interface-definition-library-method-wizard.md:45]: should be 'a' not 'an'  It's a weird case but based on the pronunciation it should be 'a'

File: docs/standard-library/integer-sequence-class.md [hunk 5]
```diff
@@ -106,7 +106,7 @@ To make an `index_sequence` for a parameter pack, use `index_sequence_for`\<T...
 
 Header: \<type_traits\>
 
-Namepace: std
+Namespace: std
```
Comment (TylerMSFT) [inline docs/standard-library/integer-sequence-class.md:109]: While you are here, could you also code escape std?

File: docs/standard-library/valarray-class.md [hunk 6]
```diff
@@ -1962,7 +1962,7 @@ int main( )
     // value_type declaration and initialization:
     valarray<int>::value_type Right = 10;
 
-    cout << "The decalared value_type Right is: "
+    cout << "The declared value_type Right is: "
```
Comment (TylerMSFT) [inline docs/standard-library/valarray-class.md:1965]: Please update the Output block below it to match

File: docs/standard-library/not-equal-to-struct.md [hunk 7]
```diff
@@ -93,7 +93,7 @@ int main( )
    transform ( v1.begin( ), v1.end( ), v2.begin( ), v3.begin ( ),
       not_equal_to<double>( ) );
 
-   cout << "The result of the element-wise not_equal_to comparsion\n"
+   cout << "The result of the element-wise not_equal_to comparison\n"
```
Comment (TylerMSFT) [inline docs/standard-library/not-equal-to-struct.md:96]: please update the code output block below this to match

File: docs/standard-library/bitset-class.md [hunk 8]
```diff
@@ -1321,7 +1321,7 @@ int main( )
 
    bitset<5> b1r3;
```
Comment (TylerMSFT) [inline docs/standard-library/bitset-class.md:1322]: please update the code output section below here to match

File: docs/standard-library/valarray-operators.md [hunk 9]
```diff
@@ -723,7 +723,7 @@ int main( )
 
    vaNE = ( vaL < vaR );
    cout << "The element-by-element result of "
-        << "the less-than comparson test is the\n"
+        << "the less-than comparison test is the\n"
```
Comment (TylerMSFT) [inline docs/standard-library/valarray-operators.md:726]: please update the code output block below to match

**Review by TylerMSFT — CHANGES_REQUESTED** [review]
Thank you for the cleanup pass!
Left some things for you to look at.

File: docs/cppcx/collections-c-cx.md (hunk 3 above)
Comment (Blake-Madden) [reply docs/cppcx/collections-c-cx.md]: Fixed

**Review by TylerMSFT — APPROVED** [review]
Thank you so much!

**Comment by TylerMSFT** [comment]
#sign-off

**Pull request merged**